djangoProject3/DFS/serializer.py

```diff
@@ -0,0 +1,176 @@
+from rest_framework import serializers
+from django.contrib.auth.models import User
+from .models import *
+from rest_framework.validators import UniqueValidator
+from django.contrib.auth.password_validation import validate_password
+from django.contrib.auth import authenticate
+
+
+class RegisterSerializer(serializers.ModelSerializer):
+    email = serializers.EmailField(
+        required=True,
+        validators=[UniqueValidator(queryset=User.objects.all())]
+    )
+
+    password = serializers.CharField(write_only=True, required=True, validators=[validate_password])
+    password2 = serializers.CharField(write_only=True, required=True)
+
+    class Meta:
+        model = User
+        fields = ('username', 'password', 'password2', 'email', 'first_name', 'last_name')
+
+    def validate(self, attrs):
+        if attrs['password'] != attrs['password2']:
+            raise serializers.ValidationError({"password": "Password fields didn't match."})
+
+        return attrs
+
+    def create(self, validated_data):
+        user = User.objects.create(
```

Comment:
pop the password and password 2 and pass the rest of the validated data in one go instead of manually setting each field

```
password = validated_data.pop('password')
# same for password 2, and then 
user = User.objects.create(**validated_data)
```

then you can set password for user manually